Submodels/MTSOL_call.py

```
@@ -198,34 +303,85 @@ def GenerateProcess(self,
+
+            # Give the thread time to notice the shutdown event
+            wait_time = 0.01
+            max_wait = 0.5
+            start = time.monotonic()
+            while self.reader.is_alive() and (time.monotonic() - start) < 1.0:
+                time.sleep(wait_time)
+                wait_time = min(max_wait, wait_time * 2)
+            try:
+                # Only force-close stdout if the thread is still alive after waiting
+                if self.reader.is_alive() and getattr(self, "process", None):
+                    self.process.stdout.close()
```

Pylint (3.3.7)

[error] Access to member 'process' before its definition line 347

(E0203)